From 1c5e9b819ab3b53980e2db06412e146f70fca109 Mon Sep 17 00:00:00 2001
From: Wendy Liang <wendy.liang@xilinx.com>
Date: Thu, 27 Aug 2015 19:40:47 -0700
Subject: [PATCH 0842/1566] remoteproc: zynqmp_r5_remoteproc: config TCM
 before write to it

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

Without configuring TCM before writing to it, you can just write
to TCM_A, you will need to configure TCM in split mode or
lockstep mode before you can write to the full TCM used by RPU.

Signed-off-by: Wendy Liang <jliang@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
(cherry picked from commit c430491c3907b83291c8242a0ef8654ce241f4fd)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/remoteproc/zynqmp_r5_remoteproc.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/remoteproc/zynqmp_r5_remoteproc.c b/drivers/remoteproc/zynqmp_r5_remoteproc.c
index b736606..70e64a9 100644
--- a/drivers/remoteproc/zynqmp_r5_remoteproc.c
+++ b/drivers/remoteproc/zynqmp_r5_remoteproc.c
@@ -574,6 +574,7 @@ static void zynqmp_r5_rproc_init(struct rproc *rproc)
 
 	dev_dbg(dev, "%s\n", __func__);
 
+	local->rpu_ops->core_conf(local);
 	local->rpu_ops->halt(local, true);
 	local->rpu_ops->en_reset(local, false);
 }
-- 
1.7.5.4

